src/Framework/FieldsAPI/Factory.php

```diff
@@ -0,0 +1,27 @@
+<?php
+
+namespace Give\Framework\FieldsAPI;
+
+use Give\Framework\FieldsAPI\Contracts\Node;
+use Give\Framework\FieldsAPI\Exceptions\TypeNotSupported;
+
+/**
+ * @unreleased
+ */
+class Factory {
+
+	/**
+	 * @unreleased
+	 * @param string $type
+	 * @param string $name
+	 * @return Node
+	 * @throws TypeNotSupported
+	 */
+	public function make( $type, $name ) {
```

Comment:
To make this more future proof, let's do this:
```php
public function make( $type, ...$args ) {
    return new $class( ...$args );
}
```

Reply:
@JasonTheAdams resolved by 5cd1cfc3591747e0f196be86708ffa9ede73b107.